core/block.ts

```diff
@@ -1116,18 +1118,11 @@ export class Block implements IASTNodeLocation, IDeletable {
    */
   getVarModels(): VariableModel[] {
     const vars = [];
-    for (let i = 0, input; (input = this.inputList[i]); i++) {
-      for (let j = 0, field; (field = input.fieldRow[j]); j++) {
-        if (field.referencesVariables()) {
-          const model = this.workspace.getVariableById(
-            field.getValue() as string,
-          );
-          // Check if the variable actually exists (and isn't just a potential
-          // variable).
-          if (model) {
-            vars.push(model);
-          }
-        }
+    for (const id of this.getVars()) {
+      const model = this.workspace.getVariableById(id);
+      // Check if the variable exists (and isn't just a potential variable).
```

Comment:
Hmm this makes me suspicious that the other code will actually fail if there is a potential variable. I'm not sure where these methods are being called from exactly though. Did you dig into this at all? If not I can take a look!

Reply:
I think this part works the same as it did before. This is used, among others:
- getVarModels in procedures
- getVariableUsesById(id) / allUsedVarModels(ws)

Reply:
> Hmm this makes me suspicious that the other code will actually fail if there is a potential variable.

With other code, did you mean `getVars`?

I think I see what you mean. I had kept the checks as they were to keep things simple, but we could:
- Use the "safer" `getVarModels()` instead of `getVars` where needed (or everywhere).
- Have `getVars` check that they are in the workspace just like `getVarModels()`, but this is a slight breaking change.

Reply:
I have a few other PRs to dig into so I will circle back on this later today!